include/utils/types.hpp

```diff
@@ -187,9 +187,59 @@ NLOHMANN_JSON_NAMESPACE_END
 #define EVCALLBACK(function) [](auto&& PH1) { function(std::forward<decltype(PH1)>(PH1)); }
 
 namespace Everest {
+
+enum class CmdEvent {
+    MessageParsingFailed,
+    SchemaValidationFailed,
+    HandlerException,
+    Timeout,
+    Shutdown
+};
+
+struct CmdResultError {
+    CmdEvent event;
+    std::string msg;
+    std::exception_ptr ex;
+};
+
+struct CmdResult {
+    std::optional<json> result;
+    std::optional<CmdResultError> error;
```

Comment:
to my understanding they are mutually exclusive - can we express that through the type system, e.x a variant?

Reply:
A variant here might a bit tricky at the moment due to `CmdResultError` being convertible to `json` and result being a `json` type. But let's keep this in mind during the upcoming refactorings in this part of the code